src/Substrait.Core/Expression/RelExpression.cs

```diff
@@ -0,0 +1,14 @@
+﻿using Substrait.Relation;
+
+namespace Substrait.Expression
+{
+  /// <summary>
+  /// Base type for all relational expressions, <see cref="Protobuf.Expression"/>
+  /// </summary>
+  abstract public class RelExpression
```

Comment:
Why `RelExpression` and not just `Expression`?  An expression is not an instance or a specialization of a `Rel`

Reply:
Ah, sorry for the confusion. I called it so to mean `relational expression`. I didn't use `Expression` or `Rex` because it was hurting code readability. Like my earlier comment, I can fix it later next week or in a subsequent PR if this one is merged. Thanks !